### PR TITLE
Feature/bl 138 mentor matching component

### PR DESCRIPTION
**Pull request by UlisesGatica**
## Description

We completely scraped out the accordion and made it functional to where you can assign a mentor to a mentee. We decided to go a different route and have a modal pop up with a table with all the mentor's names, info, stack, notes, and a number of mentees assigned to that mentor. We would have loved to add more functionality to this but due to being limited with the back end, we had to work with what we got and have someone else finish the job when there is an improved back end.

Fixes # Made it so you can assign several mentors to a mentee all at once and make it look better visually 

## Loom Video

https://www.loom.com/share/119533d40d3e463dbd93c78a21d89138

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by UlisesGatica** [comment]
just waiting on the loom video now

**Review by Charles-Merrill-Johnson — APPROVED** [review]
Great job. I see no issues, Explanation video was great. Good idea to switch to the modal I think that fits better with the UI/UX design as well.

**Comment by farhaannishtar** [comment]
Nice work saving mentor matching data to state. We can now work on updating these changes to the back end. I don’t see any issues.

**Review by farhaannishtar — APPROVED** [review]

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 1]
```diff
@@ -1,22 +1,71 @@
...
 import axiosWithAuth from '../../../utils/axiosWithAuth';
-import MentorMenteeInfo from './MentorMenteeInfo';
-import { Table, Tag } from 'antd';
+import { Table, Tag, Button } from 'antd';
+import MatchingModal from './MatchingModal';
 
 const MentorMenteeMatching = () => {
   const [assignments, setAssignments] = useState([]);
+  const [modal, setModal] = useState({ show: false, data: null });
+  const [selectedMentorKeys, setSelectedMentorKeys] = useState([]);
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:9]: Good job keeping the shape of your data in mind when initializing a slice of state!

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 2]
```diff
@@ -1,22 +1,71 @@
...
   useEffect(() => {
-    const getAssignments = () => {
-      axiosWithAuth()
-        .get('/assignments')
-        .then(res => {
-          setAssignments(res.data);
-        });
-    };
-    getAssignments();
+    // assignments in this sense means assigned mentees
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:12]: Good job anticipating and addressing a potential point of confusion with a concise clarifying comment!

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 3]
```diff
@@ -1,22 +1,71 @@
...
+  // I wasn't sure if this 'data mapping' was necessary so I kept it in for now
+  const conformData = data => {
+    return data.map(p => {
+      return {
+        key: p.profile_id,
+        name: `${p.first_name} ${p.last_name}`,
+        contact: p.email,
+        stack: splitTechStack(p.tech_stack),
+        description: 'Description goes here',
+        tags: p.matched ? 'Matched' : 'Unmatched',
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:60]: Nice, concise ternary to make use of a boolean!

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 4]
```diff
@@ -1,22 +1,71 @@
...
+        email: p.email,
+        location: p.location,
+        company: p.company,
+        matchedMentors: p.selectedMentorKeys || [],
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:64]: Nice use of the logical OR operator!

File: src/components/pages/MentorMenteeMatching/MatchingModal.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    >
+      <Row>
+        <Col span={8}>
+          <p>{modal.data?.contact}</p>
+        </Col>
+        <Col span={8}>
+          <p>{modal.data?.location}</p>
+        </Col>
+        <Col span={8}>
+          <p>{modal.data?.stack}</p>
```
Comment (albert-szabo) [inline src/components/pages/MentorMenteeMatching/MatchingModal.js:27]: This is good, astute usage of the optional chaining operator throughout this section.

**Review by albert-szabo — DISMISSED** [review]
These are good changes to add improved functionality on our road to MVP. The attention to detail is evident and appreciated.

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 6]
```diff
@@ -1,22 +1,71 @@
...
+  const showModal = mentee => {
+    setModal({ show: true, data: mentee });
+    setSelectedMentorKeys(mentee.matchedMentors);
+  };
+
+  const handleCancel = () => {
+    resetModal();
+  };
+
+  const handleSave = (menteeId, selectedMentorKeys) => {
```
Comment (khaleel737) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:34]: Code is super clean and the handle changes are great showing each event and the set of changes it does, we can display an error message if it results to an unsuccessful change as a suggestion, overall really great work

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 7]
```diff
@@ -29,55 +78,32 @@ const MentorMenteeMatching = () => {
...
   // adds commas in-between the programming languages
-  const splitTechStack = obj => {
-    let string = '';
-    for (let i = 0; i < obj.tech_stack.length; i++) {
-      if (i === obj.tech_stack.length - 1) {
-        string += `${obj.tech_stack[i]}`;
-      } else string += `${obj.tech_stack[i]}, `;
-    }
-    return string;
+  const splitTechStack = techStackArray => {
```
Comment (khaleel737) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:94]: great clean code, thats saving 8 lines of code into 2 and its much much cleaner, thank you for that.

**Review by khaleel737 — APPROVED** [review]
the new matching component looks not just great but amazing, a huge thank you to @mcfrr3 @UlisesGatica for this accomplishment

File: src/components/pages/MentorMenteeMatching/MatchingModal.js (hunk 5 above)
Comment (Johnjx) [reply src/components/pages/MentorMenteeMatching/MatchingModal.js:27]: Agreed!

File: src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js [hunk 8]
```diff
@@ -29,55 +78,32 @@ const MentorMenteeMatching = () => {
       ),
     },
     { title: 'Stack', dataIndex: 'stack', key: 'stack' },
+    {
+      title: '',
+      dataIndex: 'name',
+      key: 'assignButton',
+      render: (text, record) => (
+        <Button size="large" type="primary" onClick={() => showModal(record)}>
```
Comment (Johnjx) [inline src/components/pages/MentorMenteeMatching/MentorMenteeMatching.js:86]: I like the new format of your table as it is similar to the implementation of the _Pending Applications_ page for Admins which contains a primary button to open the associated modal. This keeps our UX consistent across features.

**Review by Johnjx — APPROVED** [review]
Great job keeping your code modular, clear, and consistently formatted/styled throughout. Most commits were well described and I thought that the explanation for not having the sync to BE / DS was delivered better in the loom video than your description. Awesome work with Ant D!

**Pull request merged**